overlord/snapstate/storehelpers_test.go

```diff
@@ -351,3 +351,110 @@ func (s *snapmgrTestSuite) TestInstallSizeErrorNoDownloadInfo(c *C) {
 	_, err := snapstate.InstallSize(st, []snapstate.MinimalInstallInfo{snapstate.InstallSnapInfo{Info: snap1}}, 0)
 	c.Assert(err, ErrorMatches, `internal error: download info missing.*`)
 }
+
+func (s *snapmgrTestSuite) TestInstallSizeWithPrereqNoStore(c *C) {
+	st := s.state
+	st.Lock()
+	defer st.Unlock()
+
+	repo := interfaces.NewRepository()
+	ifacerepo.Replace(st, repo)
+
+	s.setupInstallSizeStore()
+
+	snap1 := snaptest.MockSnap(c, `name: some-snap
+version: 1.0
+epoch: 1
+base: core
+plugs:
+  myplug:
+    interface: content
+    content: mycontent`, &snap.SideInfo{
+		RealName: "some-snap",
+		Revision: snap.R(1),
+	})
+
+	snap2 := snaptest.MockSnap(c, `name: some-snap2
+version: 1.0
+epoch: 1
+base: core
+slots:
+  myslot:
+    interface: content
+    content: mycontent`, &snap.SideInfo{
+		RealName: "some-snap2",
+		Revision: snap.R(1),
+	})
+
+	// core is already installed
+	s.mockCoreSnap(c)
+	coreSize := getInstalledSnapSize(c, st, "core")
+
+	sz, err := snapstate.InstallSize(st, []snapstate.MinimalInstallInfo{
+		snapstate.InstallSnapInfo{Info: snap1}, snapstate.InstallSnapInfo{Info: snap2}}, 0)
+	c.Assert(err, IsNil)
+	c.Check(sz, Equals, uint64(snap1.Size+snap2.Size+coreSize))
```

Comment:
Existing tests in this suite use predefined const for expected sizes (see `snap1Size`, `snap2Size` etc.) so that any unexpected size would fail; can we do the same here?

Reply:
Actually I think all snaps are locally installed as well in this test. Maybe we should have one more test where one snap is available locally and the other isn't, and we hit the store.